### DISCUSSION TITLE
Ability to Hide Emails from Team Members

### DISCUSSION THREAD

**Discussion by stnguyen90** [appwrite/appwrite#2317] [General]
Since teams are used to group permissions, I was planning on creating a Subscribers team to restrict access to my app data based on if users are paid subscribers. The problem with this is a subscriber can use the [Get Team Memberships](https://appwrite.io/docs/client/teams?sdk=web-default#teamsGetMemberships) API call to see all other subscribers name and email. Is there any way to hide emails from team members? 

Eldad brought up an idea about adding custom roles on [Discord](https://discord.com/channels/564160730845151244/564161927899512832/900264046463426580). Also, another user also brought up this concern on [Discord](https://discord.com/channels/564160730845151244/564161927899512832/900157335597613066).

**Comment by stnguyen90** [comment]
So, since I only need 1 team, the DB Refactor's enable/disable services features should suffice for solving my problem. I can disable the Teams service to prevent users from using the Get Team Memberships API while still using the Team in the back end for permissioning.

**Reply by eldadfux** [reply]
This can be a good workaround, but once an API is disabled, it will only be accessible from the Appwrite console and not from the client **and** backend APIs. I still feel custom roles is the way to go for segmenting access for different subscriptions, it should also be pretty easy to implement.

**Reply by stnguyen90** [reply]
Uh oh...ya that won't work as a workaround for me then...I've added :+1: in https://github.com/appwrite/appwrite/discussions/2475.

**Reply by stnguyen90** [reply]
Eldad commented in https://github.com/appwrite/appwrite/discussions/2475#discussioncomment-1981000:

> After thinking more and more about this. I'm pretty sure we should limit this feature to the client API only. Devs have full control on what happens on both server and functions using the API key scopes and their own source-code, no reason to block it from the server side, just like we don't block it from the Appwrite console.

So, once that gets released (maybe as early as v0.13 according to [Eldad on discord](https://discord.com/channels/564160730845151244/870534256324124692/932522627711516712)), disabling the Teams API would suffice for me.

**Reply by stnguyen90** [reply]
[Issue](https://github.com/appwrite/appwrite/issues/3269) and [PR](https://github.com/appwrite/appwrite/pull/3270).

**Comment by Meldiron** [comment]
It makes sense to me to only provide userID with **any** response about User. When we return email or name, we make assumptions about what the application needs. In my opinion, whenever we talk about User, we should only return his ID. Then, we should have a powerful enough tool to define profiles for Users that can hold public information - this could be either new ProfilesAPI or improved UserPrefs.

---

I have heard ideas of custom roles on Teams with different issues too. If you use the team as an actual team (not a permission group), you will end up needing multiple roles, which does not make sense to define as a new team... Example:

I have a tournament website and users can create football teams. When I create a team, I am an owner. When I invite a user, he will be a member. Well, now I need to add a system of offensive and defensive players, backup players, maybe even trainers or sponsors. All of that is related to my team so should be part of 1 team.

**Reply by stnguyen90** [reply]
> I have heard ideas of custom roles on Teams with different issues too. If you use the team as an actual team (not a permission group), you will end up needing multiple roles, which does not make sense to define as a new team... Example:
> 
> I have a tournament website and users can create football teams. When I create a team, I am an owner. When I invite a user, he will be a member. Well, now I need to add a system of offensive and defensive players, backup players, maybe even trainers or sponsors. All of that is related to my team so should be part of 1 team.

Wouldn't that be additional team roles? Which I believe is possible already. Ie. `team:123/sponsor`

**Comment by stnguyen90** [comment]
In 1.4, we added something that might be perfect for this use-case: [User Labels](https://appwrite.io/docs/authentication-management#labels). With user labels, you can grant access without using a team that may expose emails.

**Reply by suleimanas** [reply]
Perfect, the only issue it's limited to 5 and only alphanumeric characters are allowed
we may need to add more in some projects, and it would be great if the allowed characters: alphanumeric, non-leading hyphen, underscore, period.
